cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -22,7 +22,13 @@ Function viableCallable(CallInstruction call) {
   )
   or
   // Virtual dispatch
-  result = call.(VirtualDispatch::DataSensitiveCall).resolve()
+  result = call.(VirtualDispatch::DataSensitiveCall).resolve() and
+  (
+    call.getNumberOfArguments() <= result.getEffectiveNumberOfParameters() and
+    call.getNumberOfArguments() >= result.getEffectiveNumberOfParameters()
+    or
+    result.isVarargs()
+  )
```

Comment:
When you add the new check here in `viableCallable`, it's used on both of the two subclasses of `DataSensitiveCall`. But isn't it only `DataSensitiveExprCall` that needs the extra check? If so, you can move it to the `resolve` predicate in that class in order to do less work.

Reply:
That's a good point. I will do that!

Reply:
Do performance and tuple counts still look good after this change?

Reply:
I ran it on `php/php-src` locally, and the tuple count still looks good, yes.